bazel/workspace_status.sh

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env sh
+CHANGED_FILES=$(git --no-pager diff --name-only origin/master $(git rev-parse --abbrev-ref HEAD))
+GIT_SHA=$(git rev-parse HEAD)
+GIT_USER=$(git config --get user.name 2>&1 || true)
```

Comment:
as per in-person chat ... can you verify this actually works in CI / CD? My expectation is that this would not work, as nobody created the `git config` with `git config --set ...`.

Reply:
so, confirmed, GIT_USER, GIT_EMAIL and USER probably won't work from CI/CD. See:
https://console.cloud.google.com/cloud-build/builds;region=global/f0322a3b-a45f-4cd4-8b37-82b9f08c7df9?project=cloud-build-290921

In terms of information that we want... 
1) we want to tell who has built this binary
2) if, at time of build, there were local changes (eg, non committed modifications, unexpected branch)
3) on which commit / branch the build is based on.

Proposal:
```
# Person building the binary. If unset, assume a generic builder.
USER=${USER:-builds@enfabrica.net}

# Is this building master? Are there local changes?
GIT_BRANCH="$(git branch --show-current)"
GIT_CHANGES="$(git status --porcelain | paste -sd "," -)" # list files locally modified / staged / pending
GIT_SHA="$(git rev-parse HEAD)" # SHA of last commit in this branch
GIT_AUTHOR="$(git show -s --format='%ae' $GIT_HASH)" # Author of last commit.

# If this is master, the variables below will have the same value as the variables above.
# If this is NOT master, they will track where from master this branch is derived.
GIT_MASTER_DISTANCE="$(git log --oneline master..$GIT_BRANCH|wc -l)" # Number of commits from master.
GIT_MASTER_SHA="$(git merge-base master $GIT_BRANCH)" # SHA of last commit on master.
GIT_MASTER_AUTHOR="$(git show -s --format='%ae' $GIT_MASTER_SHA)" # Author of last commit on master.

Reply:
I've edited GIT changes to be a little more bash friendly, and added an extra variable. But most of the recommendation here is in the new updates